core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -948,6 +964,8 @@ class TimeSeriesShard(val ref: DatasetRef,
 
   private[memstore] def updatePartEndTimeInIndex(p: TimeSeriesPartition, endTime: Long): Unit = {
```

Comment:
sync with @amolnayak311 on the race condition here while writing the update. This will also affect the long term index update issue.

Reply:
Sync'd with the team. The actual partkey instance is just one, so we don't have to worry about race condition here.